src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/ExportProviderBuilder.cs

```diff
@@ -59,20 +109,69 @@ public static async Task<ExportProvider> CreateExportProviderAsync(
         // Verify we only have expected errors.
         ThrowOnUnexpectedErrors(config, catalog, logger);
 
+        RoslynLog.Logger.Log(RoslynLog.FunctionId.LSP_MEF_Cache_Built);
+
+        // Try to cache the composition.
+        _ = WriteCompositionCacheAsync(compositionCacheFile, config, logger);
```

Comment:
should probably have ReportNonFatalError tracking on this async call

Reply:
Maybe WriteCompositionCacheAsync should only catch the ioexceptions (perhaps via IOUtilities) and if anything crazy happens that is outside the bounds of normal IO exceptions, then the ReportNonFatalError can come through.